### PR TITLE
Fat 5428 create custom auth

### PR DESCRIPTION
**Pull request by Activict**

### CODE REVIEW

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            var users = DataBase.Users.GetAll();
+
+            foreach (var user in users)
+            {
+                if (user.Email == email)
+                {
+                    return true;
+                }
+            }
+            return false;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: Just to optimize the code and use the best approach you can implement easier check, for example:
**return users.Any(u => u.Email.Equals(email));**
This example can replace your code from 20 to 27 lines

File: StoreApp/Controllers/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+using StoreApp.Models.Account;
+
+namespace StoreApp.Controllers
+{
+    public class AccountController : Controller
+    {
+        private UserValidateService userValidateService;
+        private UserService userService;
+        private IMapper config;
+        public AccountController()
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs:15]: I propose to split the blocks of code. It's easier to read it.

File: StoreApp/Controllers/AccountController.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            userValidateService = new UserValidateService();
+            userService = new UserService();
+            config = new MapperConfiguration(cfg => cfg.CreateMap<UserLoginViewModel, UserDTO>()).CreateMapper();
+        }
+        
+        [HttpGet]
+        public ActionResult Registration()
+        {
+            ViewBag.User = new UserRegistrationViewModel();
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: Why do you need to create this object and set it to ViewBag.User?

File: StoreApp/Controllers/HomeController.cs [hunk 4]
```diff
@@ -1,15 +1,115 @@
...
+using System.Linq;
 
 namespace StoreApp.Controllers
 {
     public class HomeController : Controller
     {
         public ActionResult Index()
         {
+
+            //using (var db = new StoreContext())
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/HomeController.cs]: As I told you before it is not a good approach to keep commented code. If you need the code to test your API just save this code to special file which won't be saved into a commit

File: StoreApp/Views/Account/Registration.cshtml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+﻿@model StoreApp.Models.Account.UserRegistrationViewModel
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/Registration.cshtml]: Are you sure that you need to keep this code? What does it do?

File: StoreApp/Views/Account/Registration.cshtml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+<br />
+<hr />
+
+<form method="post" action="/Account/Registration">
+
+    <div>
+        <div class="col-md-2">
+            Email:
+        </div> <input type="email" name="Email" />
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/Registration.cshtml]: Need to split these blocks

File: StoreApp/Views/Account/Login.cshtml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            <div class="col-md-offset-2 col-md-10">
+                <input type="submit" value="Login" class="btn btn-success" />
+            </div>
+        </div>
+    </div>
+}
+
+<div>
+    @Html.ActionLink("Back to List", "Index")
+</div>
```
Comment (AleksandroBoko) [inline StoreApp/Views/Account/Login.cshtml]: Do you need to keep this code?

File: StoreApp/Controllers/AccountController.cs [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult Registration(UserRegistrationViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                ViewBag.Message = "Your registraition wrong!";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: Look at the message, there is the grammar mistake)

File: StoreApp/Controllers/AccountController.cs [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+                return View(model);
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/AccountController.cs]: Do you need to use a **model**?

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+            var users = DataBase.Users.GetAll();
+
+            foreach (var user in users)
+            {
+                if (user.UserName == userName)
+                {
+                    return true;
+                }
+            }
+            return false;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: The same as previous

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+            var users = DataBase.Users.GetAll();
+
+            foreach (var user in users)
+            {
+                if (user.UserName == userDTO.UserName && user.Password == userDTO.Password)
+                {
+                    return true;
+                }
+            }
+            return false;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: The same as previous

File: StoreApp.BLL/Services/UserValidateService.cs [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+            foreach (var user in users)
+            {
+                if (user.UserName == userDTO.UserName && user.Password == userDTO.Password)
+                {
+                    return user.Role;
+                }
+            }
+
+            return null;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/UserValidateService.cs]: The same as previous

File: StoreApp.DAL/EF/StoreContext.cs [hunk 13]
```diff
@@ -27,8 +27,8 @@ public class StoreDbInitializer : CreateDatabaseIfNotExists<StoreContext>
     {
         protected override void Seed(StoreContext db)
         {
-            db.Products.Add(new Product { Name = "TestProduct",  Quantity = 1 });
-            db.SaveChanges();
+            //db.Products.Add(new Product { Name = "TestProduct",  Quantity = 1 });
+            //db.SaveChanges();
```
Comment (AleksandroBoko) [inline StoreApp.DAL/EF/StoreContext.cs]: Are you sure that you need to keep this code?

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 1 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 2 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs:15]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 3 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp/Controllers/HomeController.cs (hunk 4 above)
Comment (Activict) [reply StoreApp/Controllers/HomeController.cs]: fixed

File: StoreApp/Views/Account/Registration.cshtml (hunk 5 above)
Comment (Activict) [reply StoreApp/Views/Account/Registration.cshtml]: fixed

File: StoreApp/Views/Account/Registration.cshtml (hunk 6 above)
Comment (Activict) [reply StoreApp/Views/Account/Registration.cshtml]: fixed

File: StoreApp/Views/Account/Login.cshtml (hunk 7 above)
Comment (Activict) [reply StoreApp/Views/Account/Login.cshtml]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 8 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp/Controllers/AccountController.cs (hunk 9 above)
Comment (Activict) [reply StoreApp/Controllers/AccountController.cs]: fixed

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 10 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 11 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp.BLL/Services/UserValidateService.cs (hunk 12 above)
Comment (Activict) [reply StoreApp.BLL/Services/UserValidateService.cs]: fixed

File: StoreApp.DAL/EF/StoreContext.cs (hunk 13 above)
Comment (Activict) [reply StoreApp.DAL/EF/StoreContext.cs]: fixed

**Pull request merged**